packages/dashboard/src/components/widgets/tile/index.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { PropsWithChildren } from 'react';
+
+import Box from '@cloudscape-design/components/box';
+
+import SpaceBetween from '@cloudscape-design/components/space-between';
+import Button from '@cloudscape-design/components/button';
+import { CancelableEventHandler, ClickDetail } from '@cloudscape-design/components/internal/events';
+import {
+  colorBorderDividerDefault,
+  borderRadiusBadge,
+  colorBackgroundContainerContent,
+} from '@cloudscape-design/design-tokens';
+import Icon from '@cloudscape-design/components/icon';
+
+import { DashboardWidget } from '~/types';
+import { useSelector } from 'react-redux';
+import { DashboardState } from '~/store/state';
+import { useDeleteWidgets } from '~/hooks/useDeleteWidgets';
+
+import './index.css';
+
+type DeletableTileActionProps = {
+  widget: DashboardWidget;
+};
+const DeletableTileAction = ({ widget }: DeletableTileActionProps) => {
+  const { onDelete } = useDeleteWidgets();
+
+  const handleDelete: CancelableEventHandler<ClickDetail> = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    onDelete(widget);
+  };
+
+  return <Button onClick={handleDelete} ariaLabel='delete widget' variant='icon' iconName='close'></Button>;
+};
+
+const Divider = () => <div className='horizontal-divider' style={{ backgroundColor: colorBorderDividerDefault }} />;
+
+export type WidgetTileProps = PropsWithChildren<{
+  widget: DashboardWidget;
+  title?: string;
+  removeable?: boolean;
+}>;
+
+/**
+ *
+ * Component to add functionality to the widget container
+ * Allows a user to title a widget, add click remove
+ */
+const WidgetTile: React.FC<WidgetTileProps> = ({ children, widget, title, removeable }) => {
+  const isReadOnly = useSelector((state: DashboardState) => state.readOnly);
+
+  const isRemoveable = !isReadOnly && removeable;
+  const isResizable = !isReadOnly;
+  const headerVisible = !isReadOnly || title;
+
+  return (
+    <div
```

Comment:
Can we include some ARIA handling? This element is too generic for screen readers to understand.

Reply:
This is what I am requesting changes on.

Reply:
What aria handling do you think is required for this element? The Box cloudscape component for example does not apply any aria attributes (atleast from my inspection), and is what I was originally using until more customized styling was required on the container. I can use a different html element, but by default Box is also just a div? I don't know if any other element can be more descriptive?